### ISSUE TITLE
dragging tabs to remote-ssh teminal window pastes paths with backslashes

### ISSUE THREAD

**Issue by Spongman** [microsoft/vscode#120270]
Issue Type: <b>Bug</b>

as

VS Code version: Code 1.54.3 (2b9aebd5354a3629c3aba0a5f5df49f43d6689f8, 2021-03-15T10:55:45.459Z)
OS version: Windows_NT x64 10.0.19042
Remote OS version: Linux x64 5.4.72-microsoft-standard-WSL2
Remote OS version: Linux x64 3.10.0-957.1.3.el7.x86_64
Remote OS version: Linux x64 5.4.72-microsoft-standard-WSL2

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-9400 CPU @ 2.90GHz (6 x 2904)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|47.66GB (8.94GB free)|
|Process Argv|--crash-reporter-id 70645636-42ae-4864-b371-1810b42a2956|
|Screen Reader|no|
|VM|24%|

|Item|Value|
|---|---|
|Remote|WSL: CentOS7|
|OS|Linux x64 5.4.72-microsoft-standard-WSL2|
|CPUs|Intel(R) Core(TM) i5-9400 CPU @ 2.90GHz (6 x 2904)|
|Memory (System)|37.35GB (15.48GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: dev9.corp.grid-net.com|
|OS|Linux x64 3.10.0-957.1.3.el7.x86_64|
|CPUs|Intel(R) Xeon(R) CPU E5-2686 v4 @ 2.30GHz (2 x 2299)|
|Memory (System)|3.70GB (1.04GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: CentOS7|
|OS|Linux x64 5.4.72-microsoft-standard-WSL2|
|CPUs|Intel(R) Core(TM) i5-9400 CPU @ 2.90GHz (6 x 2904)|
|Memory (System)|37.35GB (15.48GB free)|
|VM|0%|
</details><details><summary>Extensions (10)</summary>

Extension|Author (truncated)|Version
---|---|---
remote-ssh|ms-|0.65.1
remote-ssh-edit|ms-|0.65.1
remote-wsl|ms-|0.54.6
gitlens|eam|11.3.0
php-debug|fel|1.14.12
autoscrolldown|fox|1.1.1
svn-scm|joh|2.13.5
cpptools|ms-|1.2.2
edit-with-shell|ryu|1.2.0
plsql-language|xyz|1.8.2

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
pythonvspyt700cf:30270857
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492cf:30256860
vstry914:30276682
pythonvsdeb440:30248342
pythonvsded773:30248341
pythonvspyt875:30259475
pythontbcf:30265426
pythonvspyt943:30280189
wslfolderdoccf:30282075
vspre833cf:30267465
pythonptprofiler:30281270
vsdfh931cf:30280410
vshan820cf:30276953

```

</details>

**Comment by JacksonKearl** [comment]
Is the path correct for the environment, other than the path separator? (i.e. is it an absolute path the linux environment would understand if it had `/`'s)

**Comment by felipecrs** [comment]
Any hint on fixing this one? I can craft something if someone helps me get started... (I tried to look up by myself and could not find any clue).

**Comment by r3m0t** [comment]
The dragging is in `src\vs\workbench\browser\dnd.ts` in `fillEditorsDragData`, and the dropping is in `src\vs\workbench\contrib\terminal\browser\terminalInstance.ts` in `onDrop`, identify which branch is taken. `URI.file` always refers to the OS path separator (Windows in the top comment's example) so you want something more appropriate to the underlying source of the file (remote or local) - look at the implementation of the `copyFilePath` command.

**Comment by r3m0t** [comment]
Yes, it is like that. You would need to make a fake extension using https://code.visualstudio.com/api/references/contribution-points#contributes.resourceLabelFormatters to change the path separator. There's a sample I usually use https://github.com/microsoft/vscode-extension-samples/tree/main/fsprovider-sample

**Comment by felipecrs** [comment]
This issue was magically fixed... even on VSCode Stable:

https://github.com/microsoft/vscode/assets/29582865/cad0dabb-a77c-40b3-a93f-0a9071132368

I guess you can close it too.
